src_zlib/service/zlib_service.cpp

```diff
@@ -214,6 +234,163 @@ class probe
 
 //------------------------------------------------
 
+
+namespace {
+void* zalloc_impl(
+    void* /* opaque */,
+    unsigned items,
+    unsigned size)
+{
+    try {
+        return ::operator new(items * size);
+    } catch(std::bad_alloc const&) {
+        return Z_NULL;
+    }
+}
+
+void zfree_impl(void* /* opaque */, void* addr)
+{
+    ::operator delete(addr);
+}
+} // namespace
+
+class BOOST_HTTP_PROTO_ZLIB_DECL
+    zlib_filter final : public filter
```

Comment:
well, what kind of filter is it? the deflating kind or the inflating kind? Because we will need both...